content/protocols/mqtt.textile

```diff
@@ -0,0 +1,240 @@
+    return;
+  }
+});
+```
+
+h2(#pub-sub). Pub/Sub with MQTT
+
+"Ably's channels":/channels correlate to topics in MQTT. An example of how to publish and subscribe with the NodeJS "MQTT package":https://www.npmjs.com/package/mqtt would be as follows:
+
+```[nodejs]
+const mqtt = require('mqtt');
+const encoding = require('text-encoding');
+var decoder = new encoding.TextDecoder();
+var options = {
+  keepalive: 30,
+  username: '{{API_KEY_NAME}}', /* API key's name */
+  password: '{{API_KEY_SECRET}}', /* API key's secret */
+  port: 8883
+};
+
+var client = mqtt.connect('mqtts:mqtt.ably.io', options);
+
+client.on('message', function(topic, message) {
+  console.log(decoder.decode(message));
+});
+
+client.subscribe('my_channel', function (err) {
+  if (!err) {
+    client.publish('my_channel', 'my_message', { qos: 0 });
+  }
+});
+```
+
+h2(#decode-messages). Decode messages
+
+Any data published through or received by the MQTT adapter will be binary encoded, due to MQTT being a binary protocol. This means that you'll need to interpret the message to get the original contents out. For example, to interpret a message using Ably Realtime with JavaScript you'd need to do the following, using the "text-encoding NPM module's TextDecoder":https://www.npmjs.com/package/text-encoding to decode from binary to text:
+
+```[javascript] 
+var ably = new Ably.Realtime('{{API_KEY}}');
+var decoder = new TextDecoder();
+var channel = ably.channels.get('my_channel');
+
+channel.subscribe(function(message) {
+  var command = decoder.decode(message.data);
+});
+```
+
+Or, if you wish to decode messages received through MQTT, you can use the following in NodeJS:
+
+```[nodejs]
+const encoding = require('text-encoding');
+var decoder = new encoding.TextDecoder();
+var client = mqtt.connect('mqtts:mqtt.ably.io', options);
+
+client.on('connect', function () {
+  client.subscribe('my_channel');
+});
+
+client.on('message', function (topic, message) {
+  console.log(decoder.decode(message));
+});
+```
+
+In the above example, @command@ will now contain the message in its original string form.
+
+h2(#ssl-tls). SSL/TLS
+
+Ably supports both SSL and non-SSL connections (the latter uses a different port, see above), but strongly recommend using SSL wherever possible. If you are not using SSL, note that the same restrictions apply as if you were using an Ably client without SSL. That is, "connection attempts using Basic Authentication (i.e. an API key) are disallowed":/auth/basic, and any "namespaces which you've enabled 'require TLS' on":/channels will be inaccessible. "Contact us":https://ably.com/contact if this is a problem for you.
+
+h2(#channel-options). Channel options
+
+In an MQTT connection, it is also possible to specify "channel options":/channels/options as a query string in the connection URL, instead of as a qualifier on an individual channel name. In this case, the given channel options apply to all channel attachments associated with that connection.
```

Comment:
I think this is mixing MQTT and SSE a little here. 

My understanding is that for MQTT and SSE you can specify options using a query string in the channel name qualifier. SSE then also has an additional option whereby you can specify the options as part of the connection URL, however that would apply the same options to every channel you attach to with that connection. 

Might be worth double checking this!

Reply:
Confirmed with Simpson Woolf.